WordPressAuthenticator/NUX/NUXButton.swift

```diff
@@ -1,54 +1,285 @@
 import UIKit
 import WordPressShared
+import WordPressUI
+
+/// A protocol for an element that can display a UIActivityIndicatorView
+@objc public protocol ActivityIndicatorButton {
+    func showActivityIndicator(_ show: Bool)
+}
 
 /// A stylized button used by Login controllers. It also can display a `UIActivityIndicatorView`.
-@objc open class NUXButton: NUXSubmitButton {
-    // MARK: - Configuration
-    fileprivate let horizontalInset: CGFloat = 20
-    fileprivate let verticalInset: CGFloat = 12
-    fileprivate let maxFontSize: CGFloat = 22
+@objc open class NUXButton: UIButton, ActivityIndicatorButton {
+    @objc var isAnimating: Bool {
+        get {
```

Comment:
@mindgraffiti you can drop the `get  { }` block. If a property is readOnly, you can just do something like this:

```
var isAnimating: Bool {
   return activityIndicator.isAnimating
}
```

Reply:
✔️